src/json.hpp.re2c

```diff
@@ -618,7 +700,8 @@ class basic_json
         boolean,        ///< boolean value
         number_integer, ///< number value (integer)
         number_float,   ///< number value (floating-point)
-        discarded       ///< discarded by the the parser callback function
+        discarded,      ///< discarded by the the parser callback function
+        number_unsigned ///< number value (unsigned integer)
```

Comment:
Ok, I'll change the order and add from 2.0.0 to the changes.

I've solved the Valgrind problem.  It was down to documented Valgrind behaviour, which is unable to deal with floating point calculations involving `long double` and as a result it was truncating them to `double` and causing the `if(int_part != m_value.number_unsigned)` test to return true which caused the result to be stored as a `number_float_t` instead of a `number_unsigned_t`. For some reason this only happened with the unsigned case... but in any event it doesn't matter, the solution is simply to parse with `strtod` instead of `strtold`, which I should have been doing anyway.  Introducing a `long double` just introduces compiler dependant behaviour (because VS treats them as `double` by default).

While debugging that issue I came up with a better way of parsing which I am still exploring.  It  only requires one parse for integers and has a much simpler (and importantly more obvious) decision methodology.  I would have had it up and running already only I made the fateful decision to upgrade my Ubuntu installation...